ginhttp/server_test.go

```diff
@@ -153,3 +153,50 @@ func TestURLTagOption(t *testing.T) {
 		})
 	}
 }
+
+func TestPanic(t *testing.T) {
+
+	tests := []struct {
+		handler         gin.HandlerFunc
+		spanFinishCount int
+		tag             string
+	}{
+		{
+			func(c *gin.Context) {
+				c.String(http.StatusOK, "OK")
+			},
+			1,
+			"OK",
+		},
+		{
+			func(c *gin.Context) {
+				panic("panic test")
+			},
+			1,
+			"Panic",
+		},
+	}
+
+	for _, tt := range tests {
+		testCase := tt
+		t.Run(testCase.tag, func(t *testing.T) {
+			tr := &mocktracer.MockTracer{}
+			mw := Middleware(tr)
+			r := gin.New()
+			r.Use(gin.Recovery(), mw)
```

Comment:
If you reverse this to be
` r.Use(mw, gin.Recover())`

you can check for a status code of 500 in the panic case and 200 in the non-panic case.

See https://github.com/gin-gonic/gin/blob/master/recovery.go#L68